backend/httpclient/tracing_middleware.go

```diff
@@ -25,6 +25,7 @@ const (
 // code as span attributes. If tracer is nil, it will use tracing.DefaultTracer().
 func TracingMiddleware(tracer trace.Tracer) Middleware {
 	return NamedMiddlewareFunc(TracingMiddlewareName, func(opts Options, next http.RoundTripper) http.RoundTripper {
+		tracer := tracer
```

Comment:
I don't understand why this resolves the data race, it's still a pointer to the same thing, no?

Reply:
This is copying `nil` if the `tracer` does not exist, not a pointer to `nil`, so line 30 does not modify the argument `tracer`.

To make it more clear, I would change the variable name:

```
t := tracer
if t == nil {
....
```

Reply:
I have applied Andres's suggestion on changing the name, does it make more sense now? :smile:

Reply:
Sorry for slow response. Make sense

Reply:
No worries! :smile: